api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -40,6 +55,36 @@
 	@Autowired
 	private ObservationCategoryMap categoryMap;
 	
+	@Autowired
+	@Getter(AccessLevel.PUBLIC)
+	@Setter(AccessLevel.PUBLIC)
+	@Qualifier("sessionFactory")
+	private SessionFactory sessionFactory;
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
+		if (!theParams.getParameters(FhirConstants.LASTN_SEARCH_HANDLER).isEmpty()) {
+			
+			Criteria criteria = sessionFactory.getCurrentSession().createCriteria(typeToken.getRawType());
+			
+			setupSearchParams(criteria, theParams);
+			
+			criteria.setProjection(Projections.projectionList().add(Projections.property("uuid"))
+			        .add(Projections.property("concept")).add(Projections.property("obsDatetime")));
+			
+			@SuppressWarnings("unchecked")
+			List<Object[]> results = criteria.list();
+			
+			HashMap<Concept, List<Object[]>> code_groups = new HashMap<>();
+			handleGrouping(code_groups, results);
+			
+			return getLastNUuids(code_groups, getMax(theParams)).stream().distinct().collect(Collectors.toList());
```

Comment:
As a general convention for this, I'd rather we use `Lastn` rather than `LastN`, so this would be `getLastnUuids()`. I think this is a little easier to read.